src/packages/ganache/README.md

```diff
@@ -192,6 +192,59 @@ const provider = new ethers.providers.Web3Provider(ganache.provider());
 
 New RPC documentation coming soon! See https://github.com/trufflesuite/ganache/tree/master#options for Ganache v2 documentation.
 
+### Ganache Provider Events
```

Comment:
Copying from slack so we have history here:

Alternate event system ideas:
1. Instead of `context`, name it `token` or `eventToken`.
Because the whole "unique token" thing above is kinda of weird and confusing: Instead of "lifecycle events" (`ganache:vm:tx:before`, `ganache:vm:tx:step`, `ganache:vm:tx:after`) we could emit a single `ganache:vm:tx` event that contains it's own subscription methods:
```TypeScript
interface TransactionEvent {
   onBefore(): (listener: () => void)
   onStep: (listener: (event: StepEvent) => void)
   onAfter(): (listener: () => void)
}
```
alternatively, it could just be a normal event emitter (`.on`), though this necessitates some additional performance considerations (as previously described).
If we did this the user would be in charge of keeping track of each transaction's context.

Reply:
I've made this an issue for review for milestone 7.0.0 and ship the next alpha how things are now: https://github.com/trufflesuite/ganache/issues/1177